scripts/pyenv_setup.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# This script correctly sets up pyenv
+#
+# Assumptions:
+# - This script assumes you're calling from the top directory of the repository
+set -eu
+
+# Check if a command is available
+require() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+query_big_sur() {
+    if require sw_vers && sw_vers -productVersion | grep -E "11\." > /dev/null; then
+        return 0
+    fi
+    return 1
+}
+
+get_shell_startup_script() {
+  if [ -n "$SHELL" ]; then
+    case "$SHELL" in
+      /bin/bash)
+        echo "${HOME}/.bashrc"
+        ;;
+      /bin/zsh)
+        echo "${HOME}/.zshrc"
+        ;;
+    esac
```

Comment:
Should we yell at the user if they're using a diff shell?

Reply:
+1, I also saw someone wanting to use fish today or yesterday, probably add that in here while you're at it.